lib/c-api/build.rs

```diff
@@ -469,31 +469,33 @@ fn exclude_items_from_wasm_c_api(builder: Builder) -> Builder {
         .exclude_item("wasi_version_t")
         .exclude_item("wasm_config_set_compiler")
         .exclude_item("wasm_config_set_engine")
-        .exclude_item("wasm_config_set_target")
-        .exclude_item("wasm_cpu_features_add")
-        .exclude_item("wasm_cpu_features_delete")
-        .exclude_item("wasm_cpu_features_new")
-        .exclude_item("wasm_cpu_features_t")
-        .exclude_item("wasm_module_name")
-        .exclude_item("wasm_module_set_name")
-        .exclude_item("wasm_named_extern_module")
-        .exclude_item("wasm_named_extern_name")
-        .exclude_item("wasm_named_extern_t")
-        .exclude_item("wasm_named_extern_unwrap")
-        .exclude_item("wasm_named_extern_vec_copy")
-        .exclude_item("wasm_named_extern_vec_delete")
-        .exclude_item("wasm_named_extern_vec_new")
-        .exclude_item("wasm_named_extern_vec_new_empty")
-        .exclude_item("wasm_named_extern_vec_new_uninitialized")
-        .exclude_item("wasm_target_delete")
-        .exclude_item("wasm_target_new")
-        .exclude_item("wasm_target_t")
-        .exclude_item("wasm_triple_delete")
-        .exclude_item("wasm_triple_new")
-        .exclude_item("wasm_triple_new_from_host")
-        .exclude_item("wasm_triple_t")
         .exclude_item("wasmer_compiler_t")
+        .exclude_item("wasmer_config_set_compiler")
+        .exclude_item("wasmer_config_set_engine")
+        .exclude_item("wasmer_config_set_target")
```

Comment:
I mentioned it on Slack but I'll say it here for the record, I think there's a lot of value in having the string "wasm_config" appear in symbols that operate on this type to make it easier to find uses by text search.

Thus I tend to prefer `wasmer_wasm_config_*` or `wasm_config_*`.

Notably `wasmer_config_*` is the only set of functions/methods that don't follow the pattern `<type_name>_<operation>`. I think that inconsistency is enough to warrant further review, aside from my previous concerns about grepability of the identifiers.

I'm open to other ideas here too though!

Reply:
For me, `wasmer_wasm_config_*` is a non-sense. `wasm_config_*` functions aren't standard (except for `_new` and `_delete`). We are writing non-standard functions, so the prefix is `wasmer_` regarding our new rules.

We can however add `#[doc(alias = "…")]` to ease findings in the documentation's search engine. Thoughts?

Reply:
Hmm, I think that `wasmer_config_*` leads to big problems if you follow it through logically:

1. A type called `wasm_config_t` exists because it's declared by wasm.h
2. A function called `wasm_config_new` exists to operate on this type, it only has the `wasm_` prefix because of where it's declared
3. Because we're adding operations in our own namespace, `wasmer`, we then add functions as `wasmer_config_*` referring to a global, unnamespaced `config` object
4. No other types named `*_config_t` can exist, no other operations on any other config type can exist

That doesn't sound like the outcome we want. In fact it destroys the entire idea behind namespaces.

To me the only logically consistent names are `wasm_config_*` (saying it's okay for us to extend their namespace) or `wasmer_wasm_config_*` (saying it's not okay for us to extend their namespace). `wasmer_config_*` causes namespace conflicts (and is also very confusing in my opinion).

Reply:
I agree with @MarkMcCaskey point here.

My 2 cents: I'd vote for using `wasm_config_*`

Reply:
Let's go for `wasm_config_set*` then :-).